File: src/main/java/catchla/yep/loader/MessagesLoader.java
package catchla.yep.loader;

import android.content.Context;

import org.mariotaku.sqliteqb.library.Expression;
import org.mariotaku.sqliteqb.library.OrderBy;

import catchla.yep.model.Conversation;
import catchla.yep.model.Message;
import catchla.yep.model.MessageCursorIndices;
import catchla.yep.provider.YepDataStore.Messages;

/**
 * Created by mariotaku on 15/5/29.
 */
public class MessagesLoader extends ObjectCursorLoader<Message> {
    private final Conversation mConversation;

    public MessagesLoader(final Context context, final Conversation conversation) {
        super(context, MessageCursorIndices.class, Messages.CONTENT_URI, Messages.COLUMNS, null, null, null);
        mConversation = conversation;
        setSelection(Expression.and(Expression.equalsArgs(Messages.ACCOUNT_ID),
                Expression.equalsArgs(Messages.CONVERSATION_ID)).getSQL());
        setSelectionArgs(new String[]{conversation.getAccountId(), conversation.getId()});
        setSortOrder(new OrderBy(Messages.CREATED_AT, false).getSQL());
    }


}

File: src/main/java/catchla/yep/loader/MessagesLoader.kt
package catchla.yep.loader

import android.content.Context
import catchla.yep.model.Conversation
import catchla.yep.model.Message
import catchla.yep.model.MessageCursorIndices
import catchla.yep.model.MessageTableInfo
import catchla.yep.provider.YepDataStore.Messages
import org.mariotaku.sqliteqb.library.Expression
import org.mariotaku.sqliteqb.library.OrderBy

/**
 * Created by mariotaku on 15/5/29.
 */
class MessagesLoader(context: Context, private val conversation: Conversation) : ObjectCursorLoader<Message>(context, MessageCursorIndices::class.java, Messages.CONTENT_URI, MessageTableInfo.COLUMNS, null, null, null) {

    init {
        selection = Expression.and(Expression.equalsArgs(Messages.ACCOUNT_ID),
                Expression.equalsArgs(Messages.CONVERSATION_ID)).sql
        selectionArgs = arrayOf(conversation.accountId, conversation.id)
        sortOrder = OrderBy(Messages.CREATED_AT, false).sql
    }


}
